yb-voyager/cmd/root.go

```diff
@@ -156,10 +159,10 @@ func validateExportDirFlag() {
 }
 
 func lockExportDir(cmd *cobra.Command) {
-	lockFileName := ".lockfile.lck"
-	// using different lockfile as import data can be run in parallel with export data(for live migration)
-	if cmd.Use == "data" && cmd.Parent().Use == "import" {
-		lockFileName = ".importDataLockfile.lck"
+	// locking export-dir per command TODO: revisit this
+	lockFileName := fmt.Sprintf(".%s%sLockfile.lck", cmd.Parent().Use, cmd.Use)
+	if cmd.Parent().Use == "data" { //import data file case
+		lockFileName = fmt.Sprintf(".%s%s%sLockfile.lck", cmd.Parent().Parent().Use, cmd.Parent().Use, cmd.Use)
```

Comment:
We can have a common utility function that can be used to generate a string ID for each command e.g. `yb-voyager import data file` to `import-data-file`. The "command id" can then be used to generate log file names as well as lock file names.
```
func GetCommandID(c *cobra.Command) string {
	if c.HasParent() {
		p := GetCommandID(c.Parent())
		if p == "" {
			return c.Name()
		} else {
			return p + "-" + c.Name()
		}
	}
	return ""
}
```